SCORER.md

```diff
@@ -21,11 +28,278 @@ You may need to build your own scorer - your own _language model_ if:
+
+The `generate_scorer_package`, once installed via the `native client` package, is usable on _all platforms_ supported by DeepSpeech. This is so that developers can generate scorers _on-device_, such as on an Android device, or Raspberry Pi 3.
+
+To install `generate_scorer_package`, first download the relevant `native client` package from the [DeepSpeech GitHub releases page](https://github.com/mozilla/DeepSpeech/releases/tag/v0.9.3) into the `data/lm` directory. The Docker image uses Ubuntu Linux, so you should use the `native_client.amd64.cuda.linux.tar.xz` package.
+
+The easiest way to do this is to use `wget`:
+
+```
+root@dcb62aada58b:/DeepSpeech# cd data/lm
+root@dcb62aada58b:/DeepSpeech/data/lm# wget https://github.com/mozilla/DeepSpeech/releases/download/v0.9.3/native_client.amd64.cuda.linux.tar.xz
+--2021-02-26 21:45:10--  https://github.com/mozilla/DeepSpeech/releases/download/v0.9.3/native_client.amd64.cuda.linux.tar.xz
+
+native_client.amd64 100%[===================>]  13.43M  11.2MB/s    in 1.2s    
+
+2021-02-26 21:45:12 (11.2 MB/s) - ‘native_client.amd64.cuda.linux.tar.xz’ saved [14086680/14086680]
+```
+
+Next, extract the package using `tar -xvf`:
+
+```
+root@dcb62aada58b:/DeepSpeech/data/lm# tar -xvf native_client.amd64.cuda.linux.tar.xz
+libdeepspeech.so
+generate_scorer_package
+LICENSE
+deepspeech
+deepspeech.h
+README.mozilla
+```
+
+You can now generate a `ken.lm` scorer file.
+
+```
+root@dcb62aada58b:/DeepSpeech/data/lm# ./generate_scorer_package \
+  --alphabet ../alphabet.txt  \
+  --lm ../../deepspeech-data/indonesian-scorer/lm.binary
+  --vocab ../../deepspeech-data/indonesian-scorer/vocab-500000.txt \
+  --package kenlm-indonesian.scorer \
+  --default_alpha 0.931289039105002 \
+  --default_beta 1.1834137581510284
+6021 unique words read from vocabulary file.
+Doesn't look like a character based (Bytes Are All You Need) model.
+--force_bytes_output_mode was not specified, using value infered from vocabulary contents: false
+Package created in kenlm-indonesian.scorer.
+```
+
+The `kenlm-indonesian.scorer` file is stored in the `/DeepSpeech/data/lm` directory within the Docker container. Copy it to the `deepspeech-data` directory.
+
+```
+root@dcb62aada58b:/DeepSpeech/data/lm# cp kenlm-indonesian.scorer ../../deepspeech-data/indonesian-scorer/
+```
+
+```
+root@dcb62aada58b:/DeepSpeech/deepspeech-data/indonesian-scorer# ls -las
+total 1820
+  4 drwxrwxr-x 2 1000 1000   4096 Feb 26 21:56 .
+  4 drwxrwxr-x 5 1000 1000   4096 Feb 25 22:24 ..
+636 -rw-r--r-- 1 root root 648000 Feb 26 21:56 kenlm-indonesian.scorer
+488 -rw-r--r-- 1 root root 499594 Feb 24 08:05 lm.binary
+ 52 -rw-r--r-- 1 root root  51178 Feb 24 08:05 vocab-500000.txt
+```
+
+#### Using the scorer file in model training
+
+You now have your own scorer file that can be used during the model training process using the `--scorer` parameter.
```

Comment:
Technically, one will / should not re-train but just run a test set

Reply:
Ah, good call